blueprints/data-solutions/dataproc-lifecycle-via-composer/terraform/main.tf

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+provider "google-beta" {
+  project = var.project_id
+  region  = var.region
+}
+
+locals {
+  shared_vpc_project = try(var.network_config.host_project, null)
+  use_shared_vpc     = var.network_config != null
+  subnet = (
+    local.use_shared_vpc
+    ? var.network_config.subnet_self_link
+    : values(module.vpc.0.subnet_self_links)[0]
+  )
+  vpc_self_link = (
+    local.use_shared_vpc
+    ? var.network_config.network_self_link
+    : module.vpc.0.self_link
+  )
+}
+
+module "vpc" {
+  source     = "../../../../modules/net-vpc"
+  count      = local.use_shared_vpc ? 0 : 1
+  project_id = var.project_id
+  name       = "vpc"
+  subnets = [
+    {
+      ip_cidr_range      = "10.128.0.0/20"
+      name               = "subnet"
+      region             = var.region
+      secondary_ip_range = {}
+    }
+  ]
+}
+
+resource "google_compute_firewall" "default-dataproc-allow-internal" {
```

Comment:
Can you use the `net-firewall` module for this?

Reply:
Done